pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -102,15 +119,37 @@ func resourceArtifactoryLdapSetting() *schema.Resource {
 			Description: `(Optional) When set, supports paging results for the LDAP server. This feature requires that the LDAP server supports a PagedResultsControl configuration. Default value is "true".`,
 		},
 		"search_filter": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			Default:     "",
-			Description: "(Optional) A filter expression used to search for the user DN used in LDAP authentication. This is an LDAP search filter (as defined in 'RFC 2254') with optional arguments. In this case, the username is the only argument, and is denoted by '{0}'. Possible examples are: (uid={0}) - This searches for a username match on the attribute. Authentication to LDAP is performed from the DN found if successful.",
+			Type:             schema.TypeString,
+			Optional:         true,
+			Default:          "",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringMatch(regexp.MustCompile(`^.+$|^$`), "must follow valid ldap filter formatting.")),
+			Description:      "(Optional) A filter expression used to search for the user DN used in LDAP authentication. This is an LDAP search filter (as defined in 'RFC 2254') with optional arguments. In this case, the username is the only argument, and is denoted by '{0}'. Possible examples are: (uid={0}) - This searches for a username match on the attribute. Authentication to LDAP is performed from the DN found if successful.",
 		},
 		"search_base": {
-			Type:        schema.TypeString,
-			Optional:    true,
-			Default:     "",
+			Type:     schema.TypeString,
+			Optional: true,
+			Default:  "",
+			ValidateDiagFunc: func(value interface{}, path cty.Path) diag.Diagnostics {
+				v, ok := value.(string)
```

Comment:
This is the 3rd time I am seeing this code. C&P code is bad. But 3 times? 

I think you can define 1 regex that will just do the job